sqlobject/src/main/java/org/jdbi/v3/sqlobject/SqlObjectFactory.java

```diff
@@ -135,7 +135,11 @@ private Handler buildBaseHandler(Class<?> sqlObjectType, Method method) {
                             sqlMethodAnnotations));
         }
 
-        if (method.isDefault()) {
+        DefaultHandlerFactory defaultHandlerFactory = instanceConfig
+                .get(DefaultHandlerFactoryConfiguration.class)
+                .getDefaultHandlerFactory();
```

Comment:
Could we pull this up to the `attach` method, and pass down the default handler factory instead of the config registry?